src/ripple/shamap/tests/SHAMap.test.cpp

```diff
@@ -173,11 +301,14 @@ class SHAMap_test : public beast::unit_test::suite
             keys[7].SetHex ("292891fe4ef6cee585fdc6fda1e09eb4d386363158ec3321b8123e5a772c6ca8");
 
             tests::TestFamily f{beast::Journal{}};
-            SHAMap map{SHAMapType::FREE, f};
+            SHAMap map{SHAMapType::FREE, f, v};
             if (! backed)
                 map.setUnbacked ();
             for (auto const& k : keys)
+            {
                 map.addItem(SHAMapItem{k, IntToVUC(0)}, true, false);
+                map.invariants();
+            }
```

Comment:
Oh!! Well that is quite nice. Is there a way that this invariant checking could be moved out of the SHAMap class itself so its not part of the public interface? That would be ideal. But I wont hold up this PR just for that.